aspnetcore/security/authentication/configure-jwt-bearer-authentication.md

```diff
@@ -0,0 +1,370 @@
+
+```csharp
+builder.Services.AddAuthentication()
+.AddJwtBearer("some-scheme", jwtOptions =>
+{
+	jwtOptions.MetadataAddress = builder.Configuration["Api:MetadataAddress"]!;
+	jwtOptions.Authority = builder.Configuration["Api:Authority"];
+	jwtOptions.Audience = builder.Configuration["Api:Audience"];
+	jwtOptions.TokenValidationParameters = new TokenValidationParameters
+	{
+		ValidateIssuer = true,
+		ValidateAudience = true,
+		ValidateIssuerSigningKey = true,
+		ValidAudiences = builder.Configuration.GetSection("Api:ValidAudiences").Get<string[]>(),
+		ValidIssuers = builder.Configuration.GetSection("Api:ValidIssuers").Get<string[]>()
+	};
+
+	jwtOptions.MapInboundClaims = false;
+	jwtOptions.TokenValidationParameters.ValidTypes = ["at+jwt"];
+});
+```
+
+### JWT with multiple schemes
+
+APIs often need to accommodate access tokens from various issuers. Supporting multiple token issuers in an API can be accomplished by:
+
+* **Separate APIs**: Create distinct APIs with dedicated authentication schemes for each issuer.
+* [AddPolicyScheme](/dotnet/api/microsoft.aspnetcore.authentication.authenticationbuilder.addpolicyscheme): This method can define multiple authentication schemes and implement logic to select the appropriate scheme based on token properties (e.g., issuer, claims). This approach allows for greater flexibility within a single API.
+
+```csharp
+services.AddAuthentication(options =>
+{
+	options.DefaultScheme = "UNKNOWN";
+	options.DefaultChallengeScheme = "UNKNOWN";
+
+})
+.AddJwtBearer(Consts.MY_AUTH0_SCHEME, options =>
+{
+	options.Authority = "https://dev-damienbod.eu.auth0.com/";
+	options.Audience = "https://auth0-api1";
+	options.TokenValidationParameters = new TokenValidationParameters
+	{
+		ValidateIssuer = true,
+		ValidateAudience = true,
+		ValidateIssuerSigningKey = true,
+		ValidAudiences = Configuration.GetSection("ValidAudiences").Get<string[]>(),
+		ValidIssuers = Configuration.GetSection("ValidIssuers").Get<string[]>()
+	};
+})
+.AddJwtBearer(Consts.MY_AAD_SCHEME, jwtOptions =>
+{
+	jwtOptions.MetadataAddress = Configuration["AzureAd:MetadataAddress"]; 
+	jwtOptions.Authority = Configuration["AzureAd:Authority"];
+	jwtOptions.Audience = Configuration["AzureAd:Audience"]; 
+	jwtOptions.TokenValidationParameters = new TokenValidationParameters
+	{
+		ValidateIssuer = true,
+		ValidateAudience = true,
+		ValidateIssuerSigningKey = true,
+		ValidAudiences = Configuration.GetSection("ValidAudiences").Get<string[]>(),
+		ValidIssuers = Configuration.GetSection("ValidIssuers").Get<string[]>()
+	};
+})
+.AddPolicyScheme("UNKNOWN", "UNKNOWN", options =>
+{
+	options.ForwardDefaultSelector = context =>
+	{
+		string authorization = context.Request.Headers[HeaderNames.Authorization];
+		if (!string.IsNullOrEmpty(authorization) && authorization.StartsWith("Bearer "))
+		{
+			var token = authorization.Substring("Bearer ".Length).Trim();
+			var jwtHandler = new JwtSecurityTokenHandler();
```

Comment:
Can we update this to use the `JsonWebTokenHandler`? It's faster and what's now used by default internally by the JWT and OIDC auth handers.

Reply:
Updated to JsonWebTokenHandler

Reply:
And ReadJsonWebToken